src/pages/_error.tsx

```diff
@@ -0,0 +1,22 @@
+import { type NextPage, type NextPageContext } from 'next'
+
+interface ErrorProps {
+  statusCode?: number
+}
+
+const Error: NextPage<ErrorProps> = ({ statusCode }) => {
+  return (
+    <p>
+      {statusCode
+        ? `An error ${statusCode} occurred on server`
+        : 'An error occurred on client'}
+    </p>
+  )
+}
+
+Error.getInitialProps = ({ res, err }: NextPageContext) => {
+  const statusCode = res ? res.statusCode : err ? err.statusCode : 404
```

Comment:
```typescript
if (!res && !err) return 404;
const statusCode = res?.statusCode ?? err?.statusCode
```
3항 두개보다 이건어떨까요

Reply:
```typescript
if (res && "status" in res ) return res.status;
return err ? err.status : 404;
```

Reply:
어찌됐든 3항 두번 보다는 풀어서쓰는게 좋을거같아요

Reply:
```suggestion
  if (!res && !err) return { statusCode: 404 }
  const statusCode = res?.statusCode ?? err?.statusCode
  return { statusCode }
}
```

ErrorProps 인터페이스로 인한 타입 에러 발생으로 위와 같은 로직으로 개선하였습니다